XML files containing a prolog header should start with "<?xml" characters (xml:S1778)
Intentionality issue | Not complete Reliability What is clean code?
Why is this an issue?
The prolog header is the following piece of code that some XML documents start with:
<?xml version="y.x" encoding="zzzzz"?>
When the prolog exists in an XML document, it should be at the beginning of the file to enable programs to determine the encoding of non-UTF-8,
Noncompliant code example
<!-- Generated file --> <!-- Noncompliant
<?xml version="1.0" encoding="UTF-8"?>
<firstNode>
content
</firstNode>